salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +50,94 @@ def node_info(node, ca_minion):
     return result
 
 
+def cluster_version_info(namespace_information):
+    cluster_version = None
+
+    if namespace_information:
+        ns_info_dict = namespace_information.to_dict()
+        annotation_info = ns_info_dict.get(
+            'metadata', {}).get('annotations')
+        if annotation_info:
+            cluster_version = annotation_info.get(
+                'metalk8s.scality.com/cluster-version'
+            )
+    if not cluster_version:
+        cluster_version = __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return cluster_version
+
+
+def get_storageclass(storageClassV1):
+    storage_classes = {}
+    storageclass_list = storageClassV1.list_storage_class()
+    if not storageclass_list:
+        return {}
+    for storageclass in storageclass_list.items:
+        storage_classes[storageclass.metadata.name] = storageclass
+
+    return storage_classes
+
+
+def volume_info(customObjectApi, minion_id, storageclassObject):
```

Comment:
Ditto, let's go for something like:

```python
def list_volumes(api_client, node_name):
    customObjectApi = kubernetes.client.CustomObjectsApi(api_client=api_client)

    # NOTE: I'd be wary of API errors when listing storage classes as well
    storage_classes = list_storage_classes(api_client)

    try:
        volume_list = customObjectApi.list_cluster_custom_object(
            group="storage.metalk8s.scality.com",
            version="v1alpha1",
            plural="volumes",
        )
    except ApiException as exc:
        return __utils__['pillar_utils.errors_to_dict'](...)

    return dict(
        (volume.metadata.name, volume_info(volume, storage_classes))
        for volume in volume_list
    )
```

And add a `volume_info` helper to generate the dict you want from a `Volume` object and a dict of `StorageClass`es.